pkg/kubelet/util/tracelog/trace_log.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracelog
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// TraceLog keeps track of a set of log "modules" and allows us to log
+// with a specific frequency
+// TODO: move this package to kubernetes/utils
+type TraceLog struct {
+	name            string
+	traceLogModules map[string]*time.Time
+	frequency       time.Duration
+	lock            sync.RWMutex
```

Comment:
lock should be placed with the protected object
type TraceLog struct {
	name      string
	frequency time.Duration
	logger    func(msg string, keysAndValues ...interface{})

	lock            sync.RWMutex
	traceLogModules map[string]*time.Time
}